File: lua/numbers/spec/numbers_recursive_spec.lua
local nr = require("numbers_recursive")

describe("Rescursive Specs", function()
  it("fibonacci", function()
    assert.are.equal(5, nr.fibonacci(5))
  end)

  it("factorial", function()
    assert.are.equal(120, nr.factorial(5))
  end)

  it("sum_numbers", function()
    assert.are.equal(15, nr.sum_numbers(5))
  end)
end)

File: haskell/numbers/spec/NumbersRecursiveSpec.hs
module NumbersRecursiveSpec (spec) where

import Test.Hspec
import NumbersRecursive (fibonacci, factorial, sumNumbers)

spec :: Spec
spec = do
  describe "fibonacci" $ do
    it "calculates the nth Fibonacci number" $ do
      fibonacci 5 `shouldBe` 5
  
  describe "factorial" $ do
    it "calculates factorial of a number" $ do
      factorial 5 `shouldBe` 120

  describe "sumNumbers" $ do
    it "sums numbers from 1 to n" $ do
      sumNumbers 5 `shouldBe` 15